### ISSUE TITLE
Extend some drv_flds_in options for Leung_2023

### ISSUE THREAD

**Issue by ekluzek** [ESCOMP/CMEPS#407]
After #353 is done the namelist will be extended to allow the Leung_2023 method to be used. The main development of this will be in #353, this will just enable the namelist to be turned on to use the code developed there.

dust_emis_method will have Leung_2023 as an option
zender_soil_erod_source will have the none option allowed, and to be the only choice when Leung_2023 is used

This will require only some development in the buildnml and namelist defaults to allow the new options and the error checking for it. So will not be a big effort.

**Comment by ekluzek** [comment]
I'm thinking I might NOT do this, and just allow any settings in CMEPS, but have checking in CTSM and also CAM for validity. So I might close this as a wontfix, but will decide after #353

**Comment by ekluzek** [comment]
I'm going to drop this one and close as a wontfix. CMEPS will allow the dust settings to be whatever. CTSM and CAM will then be responsible to check that can use the options given.
